crates/controller/src/hud/minimap/interaction.rs

```diff
@@ -105,15 +119,62 @@ fn click_handler(
     }
 }
 
-fn move_camera_system(
-    mut click_events: EventReader<MinimapClickEvent>,
+fn move_camera_click_system(
+    window_query: Query<&Window, With<PrimaryWindow>>,
+    mut input_events: EventReader<MouseButtonInput>,
+    hud: HudNodes<With<MinimapNode>>,
+    bounds: Res<MapBounds>,
+    mut dragging: ResMut<DraggingCamera>,
     mut camera_events: EventWriter<MoveFocusEvent>,
 ) {
-    for click in click_events.iter() {
-        if click.button() != MouseButton::Left {
+    let Some(cursor) = window_query.single().cursor_position() else {
+        return;
+    };
+
+    for event in input_events.iter() {
+        if event.button != MouseButton::Left {
+            continue;
+        }
+
+        if !event.state.is_pressed() {
+            if dragging.0 {
+                dragging.0 = false;
+            }
             continue;
         }
-        camera_events.send(MoveFocusEvent::new(click.position()));
+
+        if let Some(relative) = hud.relative_position(cursor) {
+            dragging.0 = true;
+            let proportional = Vec2::new(relative.x, 1. - relative.y);
+            let event = MoveFocusEvent::new(bounds.rel_to_abs(proportional));
+
+            camera_events.send(event);
+        }
+    }
+}
+
+fn move_camera_drag_system(
+    window_query: Query<&Window, With<PrimaryWindow>>,
+    mut input_events: EventReader<MouseMotion>,
+    hud: HudNodes<With<MinimapNode>>,
+    bounds: Res<MapBounds>,
+    mouse_buttons: Res<Input<MouseButton>>,
+    dragging: Res<DraggingCamera>,
+    mut camera_events: EventWriter<MoveFocusEvent>,
+) {
+    input_events.clear();
+    if !mouse_buttons.pressed(MouseButton::Left) || !dragging.0 {
```

Comment:
I think that we do not need `dragging` resource at all. Instead we can simplify this by having a `Local<Vec2>` storing the last mouse position and sending the drag event when the current mouse position is different to the last one while `mouse_buttons.pressed(MouseButton::Left)`.

Reply:
We need to be able to tell whether the click started inside the minimap, so that we don't react to a drag that was started outside of it. In fact, if we generalize the drag events as well, I believe we need to store a list/set of the held buttons to keep track of each one separately. What do you think?